packaging/win32/freeze_program.py

```diff
@@ -11,12 +13,19 @@
 from pathlib import Path
 
 
+# Fully-qualified path for the executable should be used with subprocess to
+# avoid unreliability (especially when running from within a virtualenv)
+python = shutil.which("python")
+if not python:
+    raise SystemExit("Cannot find python command :(")
```

Comment:
We don't find python, in a python script ?

Reply:
You would be surprised ^^

The python command may not be the one that has been used to start this script (it could be python3, or pypy)
There is a function in the stdlib to retrieve the command that started the script (hence retrieving *in theory* a python interpreter... unless we are in the very unlikely case the script has been packaged into a binary using e.g. pyinstaller)

So in the end I find it simpler and more readable to treat python as a regular external command